app/uk/gov/hmrc/gform/formtemplate/FormComponentMaker.scala

```diff
@@ -322,7 +322,7 @@ class FormComponentMaker(json: JsValue) {
 
   private lazy val hmrcTaxPeriodOpt: Opt[HmrcTaxPeriod] = {
     val oHmrcTaxPeriod: Opt[HmrcTaxPeriod] = (idType, idNumber, regimeType) match {
-      case (Some(a), Some(b), Some(c)) => HmrcTaxPeriod(IdType(a), IdNumber(b), RegimeType(c)).asRight
+      case (Some(a), Right(Some(b: TextExpression)), Some(c)) => HmrcTaxPeriod(IdType(a), b, RegimeType(c)).asRight
```

Comment:
What about all other cases? We are not handling them. I've never saw this from the compiler:

```
[warn]  [E3] app/uk/gov/hmrc/gform/formtemplate/FormComponentMaker.scala:324:46:
[warn]       Exhaustivity analysis reached max recursion depth, not all missing cases are reported.
[warn]       (Please try with scalac -Ypatmat-exhaust-depth 40 or -Ypatmat-exhaust-depth off.)
[warn]           val oHmrcTaxPeriod: Opt[HmrcTaxPeriod] = (idType, idNumber, regimeType) match {
```